src/iosrv/profile.c

```diff
@@ -347,9 +360,11 @@ srv_profile_start(struct srv_profile **sp_p, char *path, int avg, int op_cnt,
 
 		strcpy(sp->sp_dir_path, path);
 	}
+
 	sp->sp_stop = 0;
 	sp->sp_avg = avg;
 	sp->sp_id = tgt_id;
+
 	/* TODO: this ULT probably need in a lower priority ULT */
 	rc = dss_ult_create(srv_profile_dump, sp, DSS_ULT_MISC,
 			    tgt_id, 0, &sp->sp_dump_thread);
```

Comment:
Profiling on VOS xstream, dumping on helper xstream, they'll operate on shared lists and other data structures, fix the race in a follow-on patch?

Reply:
yes, I will do that in the following patch.